youtube_dl/extractor/zattoo.py

```diff
@@ -0,0 +1,261 @@
+        postdata = {
+            'client_app_token': app_token,
+            'uuid': uuid,
+            'lang': 'en',
+            'app_version': app_version,
+            'format': 'json',
+        }
+        request = sanitized_Request(
+            urljoin(self._HOST_URL, '/zapi/v2/session/hello'),
+            urlencode_postdata(postdata))
+        response = self._request_webpage(
+            request, video_id, 'Say hello')
+
+        cookie = response.headers.get('Set-Cookie')
+        session_id = self._search_regex(
+            r'beaker\.session\.id\s*=\s*(.+?);', cookie, 'session id')
+        return session_id
+
+    def _generate_cookie(self, uuid, session_id, pzuid=None):
+        if not pzuid:
+            return 'uuid=%s; beaker.session.id=%s' % (uuid, session_id)
+        return 'uuid=%s; beaker.session.id=%s; pzuid=%s' % (uuid, session_id, pzuid)
+
+    def _get_channels_display_cid(self, login_info, video_id):
+        request_url = urljoin(self._HOST_URL,
+                              '/zapi/v2/cached/channels/%s&details=False' % login_info['powerhash'])
+        data = self._download_json(
+            request_url, video_id, 'Downloading available channel list')
+        display_cid = {}
+        for elem in data['channel_groups']:
+            for channel in elem['channels']:
+                display_cid[channel['display_alias']] = channel['cid']
+        return display_cid
+
+    def _extract_cid(self, login_info, video_id, channel_name):
+        display_cid = self._get_channels_display_cid(login_info, video_id)
+        return display_cid[channel_name]
+
+    def _extract_cid_and_video_info(self, video_id):
+        data = self._download_json(
+            urljoin(self._HOST_URL,
+                    '/zapi/program/details?program_id=%s&complete=True' % video_id),
+            video_id, 'Downloading video information')
+
+        info_dict = {
+            'id': video_id,
+            'title': data['program']['title'],
+            'description': data['program'].get('description'),
+            'thumbnail': data['program'].get('image_url')
+        }
+        cid = data['program']['cid']
+        return cid, info_dict
+
+    def _store_dash_format_properties(self, dash_formats, store_list):
+        for form in dash_formats:
+            if form.get('format_note') == 'DASH video':
+                store_list.append(
+                    {
+                        'ext': form.get('ext'),
+                        'width': form.get('width'),
+                        'height': form.get('height'),
+                        'tbr': form.get('tbr'),
+                        'fps': form.get('fps'),
+                    }
+                )
+
+    def _add_stored_information(self, store_list, hls_formats, i_dash):
+        for form in hls_formats:
+            if i_dash < len(store_list):
+                form.update(store_list[i_dash])
+                i_dash += 1
+        return i_dash
```

Comment:
This looks like some overcomplicated way to say something similar to:
https://github.com/rg3/youtube-dl/blob/b0f4331002798522621aff55deaf18406d17e081/youtube_dl/extractor/gamespot.py#L83-L94